test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectHandlerTests.cs

```diff
@@ -443,12 +275,12 @@ private static TestServer CreateServer(IOptions<OpenIdConnectAuthenticationOptio
             );
         }
 
-        private static TestServer CreateServer(CustomConfigureOptions configureOptions, ILoggerFactory loggerFactory)
+        private static TestServer CreateServer(CustomConfigureOptions configureOptions, IUrlEncoder encoder, ILoggerFactory loggerFactory, OpenIdConnectAuthenticationHandler handler = null)
```

Comment:
Something that looks like this should let you replace everything you need via the service collection

https://github.com/aspnet/Identity/blob/1a74ca6972ac13569bc348672aefd19502e147e3/test/Microsoft.AspNet.Identity.InMemory.Test/FunctionalTest.cs#L157